orchestra/utils/notifications.py

```diff
@@ -83,11 +85,21 @@ def notify_status_change(task, previous_status=None):
                                assignment.worker.user.email]
         }
 
-    _notify_internal_slack_status_change(task, current_worker)
-    if task.project.slack_group_id:
-        _notify_experts_slack_status_change(task, current_worker)
-
-    if message_info is not None:
+    # TODO(joshblum): Find a good abstraction for checking preferences
+    comm_type = CommunicationPreference.CommunicationType.TASK_STATUS_CHANGE
+    comm_pref = get_object_or_None(
+        CommunicationPreference,
+        worker=current_worker,
+        communication_type=comm_type.value
+    )
+
+    if comm_pref is None or comm_pref.can_slack():
+        _notify_internal_slack_status_change(task, current_worker)
```

Comment:
before the change the code would send something if `current_worker == None`, so i'm keeping that functionality for now ( @jrbotros @marcua any opinions?)